### PR TITLE
MAINT: upgrade python version (drop python 3.6) for docker dev environment

### LINKED ISSUE #13101: MAINT: Upgrade python version in docker file
The docker file for scipy dev only has python3.6, 3.7, but python 3.6 was dropped in #13081
https://github.com/scipy/scipy/blob/803588eae04af71d88e92253b698edc0f09d0ccc/tools/Dockerfile#L20-L21

So, I think it is good time to upgrade python version or ubuntu version in the docker file.

### PR DESCRIPTION
**Pull request by AtsushiSakai**
#### Reference issue

fix #13101 

#### What does this implement/fix?
I upgraded python version , dropping 3.6, and adding 3.8 in the docker file.
I upgrade ubuntu version to focal (ubuntu 20.04) in the docker file.
I updated dev docs based on it.

### CODE REVIEW

**Review by rgommers — APPROVED** [review]
Thanks @AtsushiSakai. This looks good to me. I assume it works but didn't test yet, so I'll hold off on merging.

File: tools/Dockerfile [hunk 1]
```diff
@@ -17,16 +17,25 @@
...
-FROM ubuntu:bionic
+# ubuntu focal has python 3.8 as default
+FROM ubuntu:focal
+
+# add deadsnakes ppa to install python 3.7
+RUN apt-get update && \
+    apt-get install -y software-properties-common && \
+    add-apt-repository ppa:deadsnakes/ppa
 
 RUN apt-get update && apt-get install -y \
```
Comment (andyfaff) [inline tools/Dockerfile:40]: Can you add a `--no-install-recommends` to the apt-get line?

**Review by andyfaff — APPROVED** [review]

**Comment by andyfaff** [comment]
I checked that the image builds, and that python3.7 and 3.8 are available from the docker container.

**Comment by andyfaff** [comment]
I updated the docker image on dockerhub.  @rgommers, would you like me to add anyone else to the organisation over there?

**Comment by rgommers** [comment]
>  @rgommers, would you like me to add anyone else to the organisation over there?

Not sure who wants to be added. I already have access I believe, but I've never really used it.

**Pull request merged**